### PR TITLE
ci: Husky Upgrade from `4.3.8` to `7.0.2`

### PR DESCRIPTION
**Pull request by yukinoda**
I did the husky version migration in my project as well so applied here.
#9636 should work now (Tested with dummy commits in my local machine)

ps. The latest version is ~`7.0.0`~ `7.0.2` now

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @yukinoda! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/yukinoda/cesium/blob/ci/husky-upgrade/CONTRIBUTORS.md) file!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/ci/husky-upgrade/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Review by yukinoda — COMMENTED** [review]
Ready to be reviewed

**Review by yukinoda — COMMENTED** [review]
this is ready

**Comment by ebogo1** [comment]
Thanks @yukinoda - I'll try to get to this asap. 

As a note to myself, look through this issue/PR as well: https://github.com/CesiumGS/cesium/issues/9636.

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx pretty-quick --staged
```
Comment (ebogo1) [inline .husky/pre-commit:4]: What was the reason for running `pretty-quick` with npx here instead of `npm run pretty-quick --staged`?

**Review by ebogo1 — COMMENTED** [review]
Thanks @yukinoda! Looks just about ready, and I confirmed this works locally - just one question from me.

As a note, the 7.0.0 release notes say the .husky/.gitignore file is unnecessary:
> Improve .husky/ directory structure. .husky/.gitignore is now unnecessary and can be removed.

File: .husky/pre-commit (hunk 1 above)
Comment (yukinoda) [reply .husky/pre-commit:4]: It is the recommended way by `prettier` when using husky
https://prettier.io/docs/en/precommit.html#option-2-pretty-quickhttpsgithubcomazzpretty-quick

**Comment by yukinoda** [comment]
> Improve .husky/ directory structure. .husky/.gitignore is now unnecessary and can be removed.

this is actually created when initializing husky, but let me removed it since it is unnecessary now.
also I will upgrade to version 7.0.2, to retrieve the latest fixes

**Comment by yukinoda** [comment]
> > Improve .husky/ directory structure. .husky/.gitignore is now unnecessary and can be removed.
> 
> this is actually created when initializing husky, but let me removed it since it is unnecessary now.
> also I will upgrade to version 7.0.2, to retrieve the latest fixes

@ebogo1 
seems the `husky/.gitignore` file is generated every time `husky install` is generated.
I will just upgrade the husky to the latest version

**Comment by yukinoda** [comment]
@ebogo1 
In addition, there is an audit alert
```
found 6 vulnerabilities (1 low, 3 moderate, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```
two of them can be fixed, but let me know if is not necessary so I just revert the commit 73e8c2c
```
fixed 2 of 7 vulnerabilities in 983 scanned packages
  4 vulnerabilities required manual review and could not be updated
  1 package update for 1 vulnerability involved breaking changes
  (use `npm audit fix --force` to install breaking changes; or refer to `npm audit` for steps to fix these manually)
```

**Comment by yukinoda** [comment]
> Thanks @yukinoda! Looks just about ready, and I confirmed this works locally - just one question from me.
> 
> As a note, the 7.0.0 release notes say the .husky/.gitignore file is unnecessary:
> 
> > Improve .husky/ directory structure. .husky/.gitignore is now unnecessary and can be removed.

let me correct, the `.gitignore` you see in the cesium repo is under `_` directory, for instance, `.husky/_/.gitignore`.
Before it used to be right under `.husky` folder. So I think the release notes meant about the `.gitignore` right under `.husky`.

**Comment by yukinoda** [comment]
> @ebogo1
> In addition, there is an audit alert
> 
> ```
> found 6 vulnerabilities (1 low, 3 moderate, 2 high)
>   run `npm audit fix` to fix them, or `npm audit` for details
> ```
> 
> two of them can be fixed with `npm audit fix`, but let me know if is not necessary so I just revert the commit [73e8c2c](https://github.com/CesiumGS/cesium/commit/73e8c2cbe3887dac7d4d0d179e8589afb55024de)
> 
> ```
> fixed 2 of 7 vulnerabilities in 983 scanned packages
>   4 vulnerabilities required manual review and could not be updated
>   1 package update for 1 vulnerability involved breaking changes
>   (use `npm audit fix --force` to install breaking changes; or refer to `npm audit` for steps to fix these manually)
> ```

@ebogo1 
revert commit since Travis failed

**Comment by ebogo1** [comment]
Thanks @yukinoda - I think these changes are fine; still works with 7.0.2. I wonder if we should be adding the _ directory in .husky/ to our .gitignore file (@mramato or @lilleyse)?

**Comment by mramato** [comment]
1. Go into any file and add an unused variable
2. Try to commit using `git gui`
3. Commit is accepted without husky detecting any problems.

My understanding is that this is a problem with more than just `git gui` but may affect other UX-based git tools as well.

**Comment by mramato** [comment]
and it works in husky 4.x, so this is a regression.  I'm not sure if it's because of the integration or a bug in husky 7?

**Comment by yukinoda** [comment]
@ebogo1 

> I wonder if we should be adding the _ directory in .husky/ to our .gitignore file

assuming everyone does `npm install` after this update, it seems fine to be ignored.
Since it would be created or rewritten when executing it, which runs `husky install`.

**Comment by yukinoda** [comment]
@ebogo1 let me know if you have any concerns :)

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @yukinoda! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by ggetz** [comment]
> 1. Go into any file and add an unused variable
> 2. Try to commit using git gui
> 3. Commit is accepted without husky detecting any problems.

@mramato `eslint` isn't and wasn't being run as part of the precommit hook in this repo. The above steps shouldn't result in a husky error.

I just tried commiting with the Github Desktop GUI and the formatting step ran as expected. Is there anything else preventing this PR from moving forward?

**Comment by ggetz** [comment]
This all looks good to go from my view. Thanks @yukinoda !

**Pull request merged**